cryo/dataset.py

```diff
@@ -0,0 +1,172 @@
+"""Open datasets and process them to be used by a neural network."""
+
+import json
+import os
+import numpy as np
+import h5py
+import torch
+from torch.utils.data import DataLoader, random_split
+import functools
+from PIL import Image
+
+CUDA = torch.cuda.is_available()
+
+KWARGS = {'num_workers': 1, 'pin_memory': True} if CUDA else {}
+
+
+def open_dataset(path, new_size, is_3d):
+    """Open datasets and processes data in ordor to make tensors.
+
+    Parameters
+    ----------
+    path : string,
+        Path.
+    new_size : int.
+    is_3d : boolean
+        If 2d or 3d.
+
+    Returns
+    -------
+    tensor
+        Images in black and white.
+    """
+    if not os.path.exists(path):
+        raise OSError
+    if path.lower().endswith(".h5"):
+        data_dict = h5py.File(path, 'r')
+        all_datasets = data_dict['particles'][:]
+    else:
+        all_datasets = np.load(path)
+    dataset = np.asarray(all_datasets)
+    img_shape = dataset.shape
+    n_imgs = img_shape[0]
+    new_dataset = []
+    if is_3d:
+        dataset = torch.Tensor(dataset)
+        dataset = normalization_linear(dataset)
+        dataset = dataset.reshape((len(dataset),)+(1,)+img_shape[1:])
+    else:
+        if dataset.ndim == 3:
+            for i in range(n_imgs):
+                new_dataset.append(np.asarray(Image.fromarray(
+                    dataset[i]).resize([new_size, new_size])))
+        elif dataset.ndim == 4:
+            for i in range(n_imgs):
+                new_dataset.append(np.asarray(Image.fromarray(
+                    dataset[i][0]).resize([new_size, new_size])))
+        dataset = torch.Tensor(new_dataset)
+        dataset = normalization_linear(dataset)
+        if len(img_shape) != 4:
+            dataset = dataset.reshape(
+                (img_shape[0], 1, new_size, new_size))
+    return dataset
+
+
+def normalization_linear(dataset):
+    """Normalize a tensor.
```

Comment:
same comment about backend: is this tensor specific to pytorch?

Reply:
Idea: if those functions are torch specific, I'd suggest to have the name say it explicitely, e.g.: `normalization_linear_torch()`